IdentityCore/src/oauth2/aad_base/MSIDAADTokenResponse.h

```diff
@@ -27,20 +27,41 @@
 @interface MSIDAADTokenResponse : MSIDTokenResponse
 
 // Default properties for an AAD error response
-@property (readonly) NSString *correlationId;
+@property (nonatomic, readonly, nullable) NSString *correlationId;
 
 // Default properties for an AAD successful response
-@property (readonly) NSInteger expiresOn;
-@property (readonly) NSInteger extendedExpiresIn;
-@property (readonly) MSIDClientInfo *clientInfo;
-@property (readonly) NSString *familyId;
-@property (readonly) NSString *suberror;
-@property (readonly) NSString *additionalUserId;
+@property (nonatomic, readonly) NSInteger expiresOn;
+@property (nonatomic, readonly) NSInteger extendedExpiresIn;
+@property (nonatomic, readonly, nullable) MSIDClientInfo *clientInfo;
+@property (nonatomic, readonly, nullable) NSString *familyId;
+@property (nonatomic, readonly, nullable) NSString *suberror;
+@property (nonatomic, readonly, nullable) NSString *additionalUserId;
 
 // Custom properties that ADAL/MSAL handles
-@property (readonly) NSString *speInfo;
+@property (nonatomic, readonly, nullable) NSString *speInfo;
 
 // Derived properties
-@property (readonly) NSDate *extendedExpiresOnDate;
+@property (nonatomic, readonly, nullable) NSDate *extendedExpiresOnDate;
+
+- (nullable instancetype)initWithAccessToken:(nullable NSString *)accessToken
```

Comment:
This initializer is too long. Please consider other options :)
Additionally, how would the caller know whether they should call initializer in AADTokenResponse or in the base token response? Does it mean that the caller would need to be aware of the response type?

Reply:
removed initializers and made properties read/write.